src/performance/perfcollect/perfcollect

```diff
@@ -0,0 +1,560 @@
+	local dsosWithHits=`perf buildid-list --with-hits | grep -v /tmp/perf-`
+	for dso in $dsosWithHits
+	do
+		# Build up tuples of buildid and binary path.
+		local processEntry=0
+		if [ -f $dso ]
+		then
+			local pathToBinary=$dso
+			processEntry=1
+		else
+			local buildid=$dso
+			pathToBinary=''
+		fi
+
+		# Once we have a tuple for a binary path that exists, process it.
+		if [ "$processEntry" == "1" ]
+		then
+			# Get the binary name without path.
+			local binaryName=`basename $pathToBinary`
+	
+			# Build the debuginfo file name.
+			local destFileName=$binaryName.debuginfo
+
+			# Build the destination directory for the debuginfo file.
+			local currentDir=`pwd`
+			local destDir=$currentDir/debuginfo/$buildid
+
+			# Build the full path to the debuginfo file.
+			local destPath=$destDir/$destFileName
+
+			# Check to see if the DSO contains symbols, and if so, build the debuginfo file.
+			local noSymbols=`objdump -t $pathToBinary | grep "no symbols" -c`
+			if [ "$noSymbols" == "0" ]
+			then
+				echo "Generating debuginfo for $binaryName with buildid=$buildid"
+				mkdir -p $destDir
+				objcopy --only-keep-debug $pathToBinary $destPath
+			else
+				echo "Skipping $binaryName with buildid=$buildid.  No symbol information."
+			fi
+		fi
+	done
+
+	WriteStatus "END: Save Symbols"
+	
+	WriteStatus "START: Compress data."
+	
+	# Move all collected files to the new directory.
+	mv * $directoryName > /dev/null 2>&1
+
+	# Compress the data.
+	local archiveSuffix=".tgz"
+	local archiveName=$directoryName$archiveSuffix
+	tar -czvf $archiveName $directoryName 
+	#-C $tempDir $directoryName
+
+	# Move back to the original directory.
+	popd
+
+	# Move the archive.
+	mv $tempDir/$archiveName .
+	
+	WriteStatus "END: Compress data."
+
+	# Delete the temp directory.
+	rm -rf $tempDir	
+
+	WriteStatus "END: Archive collected data."
+}
+
+##
+# Handle the CTRL+C signal.
+##
+CTRLC_Handler()
+{
+	# The user must CTRL+C to stop collection in perf_event.
+	# When this happens, we catch the signal and finish our work.
+	ProcessCollectedData
+}
+
+##
+# Print usage information.
+##
+PrintUsage()
+{
+	echo "This script uses perf_event to collect and view performance traces."
+	echo "For detailed collection and viewing steps, view this script in a text editor or viewer."
+	echo ""
+	echo "./perfcollect <action> <tracename>"
+	echo "Valid Actions: collect view installperf"
+	echo ""
+	echo "collect options:"
+	echo "By default, collection includes CPU samples collected every ms."
+	echo "	-pid		  : Only collect data from the specified process id."
+	echo "	-threadtime       : Collect context switch events."
+	echo ""
+	echo "view options:"
+	echo "	-processfilter	  : Filter data by the specified process name."
+	echo "	-graphtype	  : Specify the type of graph.  Valid values are 'caller' and 'callee'."
+	echo ""
+	echo "installperf options:"
+	echo "	No required or optional args.  Useful for first-time setup or after kernel upgrade."
```

Comment:
Updated the text to specify "No options" since there are no options that can be specified, but I wanted a place to describe the command.